### ISSUE TITLE
Ability to nest a form field inside the label of a checkbox or radio button

### ISSUE THREAD

**Issue by mathisscott** [vmware/clarity#5891]
Results of a conversation on this issue follow below:

Following up a conversation on Slack, I would like to request the ability to design nested controls as such:

<img src=https://user-images.githubusercontent.com/113730/115374203-456fe880-a1d5-11eb-9974-cf2b8f2a1018.png width=355 />

At CloudHealth, we have a lot of form controls following this format, often having to select an option and provide 1 or 2 settings for that option. We are currently converting our platform to Clarity and encountering a lot of those patterns. Some definitely need rethinking the page/form, but in many cases it makes sense UX wise.

I was hoping that the current components would let us do something like this, but each attempt had one small thing preventing it. I’ve been trying all sorts of combinations using helper components such as `CdsInputGroup` and `CdsControl` but so far haven’t been able to make the input even just show up at all.
In a nutshell, to produce the mock above, that third row cannot be a `CdsControl` because that will only let me use 1 control (vs. 2: checkbox + input), and it cannot be a `CdsInputGroup` (which lets me have more than 1 control) because the `CdsCheckboxGroup` only fills slots with `cds-control` elements while the `CdsInputGroup` is a `cds-control-group`).
Would you be open to consider this in Clarity Core?

_Originally posted by @astorije in https://github.com/vmware/clarity/discussions/5879_

===

Of interest here is that the primary reason this is not viable right now is that the checkbox group orientation is being forced on the input inside the label. It's hard to gauge the difficulty of preventing the group from doing that. But that is the reason this won't work today.

**Comment by coryrylan** [comment]
Here is a rough workaround, you can specify the slot on the input and use margins to align it. Its not a great solution but would work until we make a supported API https://stackblitz.com/edit/typescript-1puhwe?file=index.html

**Comment by Jinnie** [comment]
Do we have any A11Y feedback on this pattern?

**Comment by coryrylan** [comment]
This should be easier to implement as a patten once i get https://github.com/vmware/clarity/issues/6053 merged. Have an example from a11y we can reference
